pkg/controller/bucketaccess/bucket_access_controller.go

```diff
@@ -98,9 +99,35 @@ func (bal *bucketAccessListener) Add(ctx context.Context, obj *v1alpha1.BucketAc
 		return nil
 	}
 
+	bucketInstanceName := obj.Spec.BucketInstanceName
+	bucket, err := bal.bucketAccessClient.ObjectstorageV1alpha1().Buckets().Get(ctx, bucketInstanceName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get bucket instance %s: %+v", bucketInstanceName, err)
+	}
+
 	req := osspec.ProvisionerGrantBucketAccessRequest{
-		BucketName: obj.Spec.BucketInstanceName,
-		Principal:  obj.Spec.Principal,
+		Principal:     obj.Spec.Principal,
+		AccessPolicy:  obj.Spec.PolicyActionsConfigMapData,
+		BucketContext: map[string]string{},
+	}
+
+	switch bucket.Spec.Protocol.Name {
+	case v1alpha1.ProtocolNameS3:
+		req.BucketName = bucket.Spec.Protocol.S3.BucketName
+		req.Region = bucket.Spec.Protocol.S3.Region
+		req.BucketContext["Version"] = bucket.Spec.Protocol.S3.Version
+		req.BucketContext["SignatureVersion"] = string(bucket.Spec.Protocol.S3.SignatureVersion)
+		req.BucketContext["Endpoint"] = bucket.Spec.Protocol.S3.Endpoint
+	case v1alpha1.ProtocolNameAzure:
+		req.BucketName = bucket.Spec.Protocol.AzureBlob.ContainerName
+		req.BucketContext["StorageAccount"] = bucket.Spec.Protocol.AzureBlob.StorageAccount
+	case v1alpha1.ProtocolNameGCS:
+		req.BucketName = bucket.Spec.Protocol.GCS.BucketName
+		req.BucketContext["ServiceAccount"] = bucket.Spec.Protocol.GCS.ServiceAccount
+		req.BucketContext["PrivateKeyName"] = bucket.Spec.Protocol.GCS.PrivateKeyName
+		req.BucketContext["ProjectID"] = bucket.Spec.Protocol.GCS.ProjectID
+	default:
```

Comment:
All the fields are expected to be populated appropriately once the createBucket is executed successfully!

Reply:
Not sure I understand this comment.  The default only comes into play if the protocol name is somehow not what we expect.